contrib/mesos/docs/scheduler.md

```diff
@@ -0,0 +1,115 @@
+# Kubernetes-Mesos Scheduler
+
+## Labels and Mesos Agent Attributes
+
+The scheduler of Kubernetes on Mesos takes [labels][1] into account: it matches
+specified labels in pod specs with defined labels of nodes.
+
+In addition to user defined labels, [attributes of Mesos agents][2] are converted
+into node labels by the scheduler, following the pattern
+
+```yaml
+k8s.mesosphere.io/attribute-<name>: value
+```
+
+As an example, an attribute `generation: 2015` will result in the node label
+
+```yaml
+k8s.mesosphere.io/attribute-generation: 2015
+```
+
+and be used to schedule pods onto node which are of generation 2015.
+
+**Note:** Node labels which originate from Mesos agent attributes must not be
+modified by the user using e.g. kubectl. Instead the Kubernetes executor will
+notice when a mesos-slave is restarted with new attributes and will modify the
+node object accordingly on the apiserver.
+
+## Resource Roles
+
+A Mesos cluster can be statically partitioned using [resources roles][2]. Each
+role is assigned such a role (`*` is the default role, if none is explicitly
```

Comment:
I think this "extra role" is what Mesos offers. It's not about inheritance. You always get the `*` resource offers. But you can subscribe to one more role.